src/tape/ensemble.py

```diff
@@ -206,6 +209,54 @@ def info(self, verbose=True, memory_usage=True, **kwargs):
         print("Source Table")
         self._source.info(verbose=verbose, memory_usage=memory_usage, **kwargs)
 
+    def check_sorted(self, table="object"):
+        """Checks to see if an Ensemble Dataframe is sorted on the index.
+
+        Parameters
+        ----------
+        table: `str`, optional
+            The table to check.
+
+        Returns
+        -------
+        A boolean value indicating whether the index is sorted (True)
+        or not (False)
+        """
+        if table == "object":
+            idx = self._object.index
+        elif table == "source":
+            idx = self._source.index
+        else:
+            raise ValueError(f"{table} is not one of 'object' or 'source'")
+        return idx.map_partitions(lambda a: np.all(a[:-1] <= a[1:])).compute().all()
```

Comment:
It is not obvious to me how this works. Doesn't map_partitions call the function (in this case the lambda) once per partition? If so, it is only checking that the last element is smaller or equal to the first one. Wouldn't that leave out the elements in the middle?

From the tests this looks like it works, so I must be missing something. Is there a comment you could add to provide more info?

Reply:
Thanks for calling this out, it's a vectorized operation that's actually doing element-wise comparison within the partition. As I was trying to put together an explanation, I realized that this is only doing comparison within the partition, and not across partitions. I've swapped this out to just use the existing `is_monotonic_increasing` function within the Dask/Pandas Index API, and expanded the test case to test the multi-partition case.